modelarmor/create_template.go

```diff
@@ -0,0 +1,105 @@
+// limitations under the License.
+
+// Sample code for creating a new model armor template.
+
+package modelarmor
+
+// [START modelarmor_create_template]
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	modelarmor "cloud.google.com/go/modelarmor/apiv1"
+	modelarmorpb "cloud.google.com/go/modelarmor/apiv1/modelarmorpb"
+	"google.golang.org/api/option"
+)
+
+// createModelArmorTemplate creates a new Model Armor template.
+//
+// This method creates a new Model Armor template with the provided settings.
+//
+// Args:
+//
+//	w io.Writer: The writer to use for logging.
+//	projectID string: The ID of the Google Cloud project.
+//	locationID string: The ID of the Google Cloud location.
+//	templateID string: The ID of the template to create.
+//
+// Returns:
+//
+//	*modelarmorpb.Template: The created template.
+//	error: Any error that occurred during template creation.
+//
+// Example:
+//
+//	template, err := createModelArmorTemplate(
+//	    os.Stdout,
+//	    "my-project",
+//	    "us-central1",
+//	    "my-template",
+//	)
+//	if err != nil {
+//	    log.Fatal(err)
+//	}
+//	fmt.Println(template)
+func createModelArmorTemplate(w io.Writer, projectID, location, templateID string) (*modelarmorpb.Template, error) {
+	ctx := context.Background()
+
+	// Create the Model Armor client.
+	client, err := modelarmor.NewClient(ctx,
+		option.WithEndpoint(fmt.Sprintf("modelarmor.%s.rep.googleapis.com:443", location)),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create client for project %s, location %s: %v", projectID, location, err)
+	}
+	defer client.Close()
+
+	// Build the Model Armor template with your preferred filters.
+	// For more details on filters, please refer to the following doc:
+	// [https://cloud.google.com/security-command-center/docs/key-concepts-model-armor#ma-filters](https://cloud.google.com/security-command-center/docs/key-concepts-model-armor#ma-filters)
+	template := &modelarmorpb.Template{
+		FilterConfig: &modelarmorpb.FilterConfig{
+			PiAndJailbreakFilterSettings: &modelarmorpb.PiAndJailbreakFilterSettings{
+				FilterEnforcement: modelarmorpb.PiAndJailbreakFilterSettings_ENABLED,
+				ConfidenceLevel:   modelarmorpb.DetectionConfidenceLevel_MEDIUM_AND_ABOVE,
+			},
+			MaliciousUriFilterSettings: &modelarmorpb.MaliciousUriFilterSettings{
+				FilterEnforcement: modelarmorpb.MaliciousUriFilterSettings_ENABLED,
+			},
+		},
+	}
+
+	// Prepare the request for creating the template.
+	req := &modelarmorpb.CreateTemplateRequest{
+		Parent:     fmt.Sprintf("projects/%s/locations/%s", projectID, location),
+		TemplateId: templateID,
+		Template:   template,
+	}
+
+	// Create the template.
+	response, err := client.CreateTemplate(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create template: %v", err)
+	}
+
+	// Print the new template name using fmt.Fprintf with the io.Writer.
+	fmt.Fprintf(w, "Created template: %s\n", response.Name)
+
+	// [END modelarmor_create_template]
```

Comment:
issue: move the ending region tag after the closing curly brace for the sample (after line 105).